backend/auth/handlers.go

```diff
@@ -1,118 +1,157 @@
+        c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{
+            "err": "Invalid credentials",
+        })
+        return
+    }
+
+    log.Printf("Succesful login as %s", credentials.Username)
+    rateLimiter.resetRateLimit(credentials.Username)
+    signedToken, err := generateAuthToken(a.jwtSecret)
+    if err != nil {
+        c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+            "err": "Failed to generate authentication token",
+        })
+        return
+    }
+
+    c.JSON(http.StatusOK, gin.H{
+        "token": signedToken,
+    })
 }
 
 // GET /refresh
 // Refresh returns a new JWT token (should be called after an auth check)
 func (a *AuthHandlers) Refresh(c *gin.Context) {
-	signedToken, err := generateAuthToken(a.jwtSecret)
-	if err != nil {
-		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
-			"err": "Failed to generate authentication token",
-		})
-		return
-	}
-
-	c.JSON(http.StatusOK, gin.H{
-		"token": signedToken,
-	})
+    signedToken, err := generateAuthToken(a.jwtSecret)
+    
+    if err != nil {
+        c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+            "err": "Failed to generate authentication token",
+        })
+        return
+    }
+
+    c.JSON(http.StatusOK, gin.H{
+        "token": signedToken,
+    })
 }
 
 // GET /single-use
 // SingleUse returns a new single-use JWT token for use in streams
 func (a *AuthHandlers) SingleUse(c *gin.Context) {
-	jti := uuid.New().String()
-
-	a.authMiddleware.setWhitelist(jti, c.Query("target"))
-
-	claims := jwt.RegisteredClaims{
-		ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Hour * 24)),
-		Issuer:    "Mikochi",
-		IssuedAt:  jwt.NewNumericDate(time.Now()),
-		ID:        jti,
-	}
-	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
-	signedToken, err := token.SignedString(a.jwtSecret)
-
-	if err != nil {
-		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
-			"err": "Failed to generate authentication token",
-		})
-		return
-	}
-
-	c.JSON(http.StatusOK, gin.H{
-		"token": signedToken,
-	})
+    jti := uuid.New().String()
+    
+
+    a.authMiddleware.setWhitelist(jti, c.Query("target"))
+
+    claims := jwt.RegisteredClaims{
+        ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Hour * 24)),
+        Issuer:    "Mikochi",
+        IssuedAt:  jwt.NewNumericDate(time.Now()),
+        ID:        jti,
+    }
+    token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+    signedToken, err := token.SignedString(a.jwtSecret)
+
+    if err != nil {
+        c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+            "err": "Failed to generate authentication token",
+        })
+        return
+    }
+
+    c.JSON(http.StatusOK, gin.H{
+        "token": signedToken,
+    })
 }
+
+// POST /logout
+// Logout invalidates the current JWT token
+func (h *AuthHandlers) Logout(c *gin.Context) {
+    authHeader := c.GetHeader("Authorization")
```

Comment:
For the sake of being DRY we can do this in JWTMiddleware.CheckAuth. I think it should just set the jti in the context

Reply:
Done